### PR TITLE
Design doc for configuration splitting for equality axioms.

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
I'll merge after the first approval.

This allows me to change this condition, which does not work for ceil-related axioms:

https://github.com/kframework/kore/blob/master/src/main/haskell/kore/src/Kore/Step/BaseStep.hs#L591

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

**Comment by virgil-serbanuta** [comment]
I added a new commit, I hope I made the text more readable.

File: docs/2019-03-06-Equality-Axiom-Configuration-Splitting.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+`g(x1)=a` and a substitution `y1=x2`.
+
+However, let us note that, unlike when rewriting, we usually don't want to
+use the most general unifier. As an example, let's say that we want to define
+`f(z)` with these axioms:
+```
+f(z) = sin(z) if z < 0
+f(z) = cos(z) if z >= 0
+```
+and let's say that we want to apply the first of these equations to `sgn(x)`.
```
Comment (traiansf) [inline docs/2019-03-06-Equality-Axiom-Configuration-Splitting.md]: isn't it here `f(x)`

File: docs/2019-03-06-Equality-Axiom-Configuration-Splitting.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+Solution
+--------
+
+Let us assume that we have several axioms `∀ Zi . αi(Zi) = βi(Zi)` which we
+think we should apply together (e.g. they form a function definition).
+
+We identify `Pi(X, Zi)` such that
+```
+(∃ Zi . Pi(X, Zi) ∧ φ(X)) = (∃ Z . Pi(X, Zi) ∧ αi(Zi)).
```
Comment (ttuegel) [inline docs/2019-03-06-Equality-Axiom-Configuration-Splitting.md]: On the right-hand side, do you mean `∃ Zi`?

**Comment by virgil-serbanuta** [comment]
Jenkins: test this please

**Comment by virgil-serbanuta** [comment]
Thanks to both, I think I have fixed your comments. Since I have an approval, I'll merge at the first green build (chances are that will not happen soon), but if you have other comments, I'll also fix those.

**Pull request merged**